docs/design/Education-organization-Endpoints.md

```diff
@@ -0,0 +1,153 @@
+  * No manual refresh endpoint - data is refreshed automatically via background
+    job
+
+* **Cross-Database Support:**
+  * Works with both SQL Server and PostgreSQL
+  * Uses C# service layer for dynamic database connections and efficient data
+    refresh
+
+## Database Schema
+
+### Tables
+
+* `adminapi.EducationOrganizations` - Stores consolidated education organization
+  data
+
+  ```sql
+  CREATE TABLE [adminapi].[EducationOrganizations] (
+        [Id] INT IDENTITY(1,1) NOT NULL,
+        [InstanceId] INT NOT NULL,
+        [InstanceName] NVARCHAR(100) NOT NULL,
+        [EducationOrganizationId] INT NOT NULL,
+        [NameOfInstitution] NVARCHAR(75) NOT NULL,
+        [ShortNameOfInstitution] NVARCHAR(75) NULL,
+        [Discriminator] NVARCHAR(128) NOT NULL,
+        [ParentId] INT NULL,
+        [OdsDatabaseName] NVARCHAR(255) NULL,
+        [LastRefreshed] DATETIME2 NOT NULL DEFAULT GETUTCDATE(),
+        [LastModifiedDate] DATETIME2 NULL,
+        CONSTRAINT [PK_EducationOrganizations] PRIMARY KEY ([Id])
+    );
+  ```
+
+## API Usage Examples
+
+### Get All Education Organizations
+
+```http
+GET /v2/educationorganizations
+Authorization: Bearer <token>
+```
+
+### Get Education Organizations for Specific Instance
+
+```http
+GET /v2/educationorganizations/123
+Authorization: Bearer <token>
+```
+
+## Configuration
+
+### Quartz.NET Job Scheduling
+
+Add to `appsettings.json`:
+
+```json
+{
+   "AppSettings": {
+   "EducationOrgsRefreshIntervalInHours": 6
+   }
+}
+```
+
+### Database Connection
+
+The system uses the existing AdminAPI database connection and dynamically
+connects to ODS instance databases based on the `OdsInstances` table
+configuration.
+
+## Architecture
+
+### Service Layer
+
+The service files can be maintained in a common project and shared between the
+V1 and V2 projects to avoid code duplication.
+
+* `IGetEducationOrganizationQuery` - Main query handling interface
+
+* `GetEducationOrganizationQuery` - Implementation of database context query logic
+  for reading the EducationOrganizations
+
+The `RefreshEducationOrganizationCommand` service layer implements a
+comprehensive C# solution to aggregate education organization data across
+multiple ODS instances and persist it into the
+[adminapi].[EducationOrganizations] table.
+
+**Key Components:**
+
+* **Dynamic Database Connectivity**: Reads ODS instance connection strings from
+    the `OdsInstances` table and establishes connections to each ODS database
+    dynamically
+* **Multi-Database Support**: Handles both SQL Server and PostgreSQL with
+    database-specific query implementations
+* **Parent Hierarchy Logic**: Implements the same parent relationship logic from
+    the AWS Lambda reference using COALESCE joins to establish education
+    organization hierarchies
+    [LambdaFunction](https://github.com/edanalytics/startingblocks_oss/blob/efc423212930e01f0166033d97be392d3a675999/lambdas/TenantResourceTreeLambdaFunction/index.mjs#L100)
```

Comment:
This function makes an interesting point about managing large numbers of databases. We do _not_ need to optimize for that large number at this time. But I can imagine 800 instances in a realistic setting.